### DISCUSSION TITLE
How is data from cpu_model.csv is used in kepler?

### DISCUSSION THREAD

**Discussion by vimalk78** [sustainable-computing-io/kepler#1031] [Q&A]
The columns in cpu_model.csv are `Model,Architecture`. sample values
```
E3-1290,Sandy Bridge
E7-8880L,Ivy Bridge
E7-4830,Haswell
D-1513N,Broadwell
```
how is the name of micro architecture used in kepler? is the name directly used as a feature in the model? or this name gets translated to some other feature set and then used?

**Comment by vimalk78** [comment]
cc: @rootfs @sunya-ch @marceloamaral @jiere

**Comment by jiere** [comment]
BTW, are another two csv files(data/power_data.csv, data/power_model.csv) still used in Kepler power modeling? If not, I will also clean them up in my [PR](https://github.com/sustainable-computing-io/kepler/pull/1029) later. Thanks.

**Comment by jiere** [comment]
There is different granularity to define CPU model, from coarse to slim:
1. Using CPU microarchitecture(i.e. Intel `Golden Cove`, which could be mapping to multiple products, see [here](https://en.wikipedia.org/wiki/List_of_Intel_CPU_microarchitectures#64-bit_(x86-64)))
2. Using CPU product codename(i.e. Intel `Sapphire Rapids`, which could be mapping to multiple products models, see[ here](https://ark.intel.com/content/www/us/en/ark/search.html?_charset_=UTF-8&q=Sapphire+Rapids))
3. Using CPU model number(i.e. 143 for Intel `Sapphire Rapids`, which is defined [here](https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/intel-family.h#L106), and could be fetched either from `cpuid` or `lscpu`) 
4. Using CPU product name(i.e. `Intel® Xeon® Platinum 8470N`, which is listed in this [url](https://ark.intel.com/content/www/us/en/ark/products/codename/126212/products-formerly-sapphire-rapids.html) and shown "8470N" in current `cpu_model.csv`)

Due to the current [script](https://github.com/sustainable-computing-io/kepler/blob/main/data/download_intel_cpu_models.sh) logic, `Model` in `cpu_model.csv` currently refers to option 4 above, `Architecture` in `cpu_mode.csv` currently not exactly refers to option 1 above, since `Sapphire Rapids` is not microarchitecture but product codename...

So let's first confirm which one of above granularity is chosen for our Kepler power modeling?

**Reply by marceloamaral** [reply]
Please, see my comment here: https://github.com/sustainable-computing-io/kepler/issues/1021#issuecomment-1788570916
We can probably just use the `CPU model name` where we can get from `/proc/cpuinfo`

**Reply by jiere** [reply]
Sorry, there is some wording misunderstanding, IIUC, what you mean is still the microarchitecture, right?
Since `CPU model name` refers to the CPU Brand name:
```
$ cat /proc/cpuinfo |grep "model name"
model name      : 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
```

I will refine the PR and use proper Golang pkg/lib instead of `lscpu` to act as alternative of  `cpuid -1 |grep uarch`
